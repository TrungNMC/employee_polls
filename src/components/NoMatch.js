import notfound from '../assets/404.png';
import NavBar from './NavBar';

function NoMatch() {
  return (
    <div className='no-macth'>
      <NavBar />
      <img
        srcSet={notfound}
        src={notfound}
        alt={'Page Not Found'}
        loading='lazy'
        width='100%'
        height='100%'
      />
    </div>
  );
}

export default NoMatch;
