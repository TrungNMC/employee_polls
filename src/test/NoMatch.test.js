import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NoMatch from '../components/NoMatch';
import React from 'react';
import { createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

const mockUsers = [
  {
    id: 'sarahedo',
    password: 'password123',
    name: 'Sarah Edo',
    avatarURL:
      'https://www.svgrepo.com/show/384671/account-avatar-profile-user-14.svg',
    answers: {
      '8xf0y6ziyjabvozdd253nd': 'optionOne',
      '6ni6ok3ym7mf1p33lnez': 'optionOne',
      am8ehyc8byjqgar0jgpub9: 'optionTwo',
      loxhs1bqm25b708cmbf3g: 'optionTwo',
    },
    questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9'],
  },
  {
    id: 'tylermcginnis',
    password: 'abc321',
    name: 'Tyler McGinnis',
    avatarURL:
      'https://www.svgrepo.com/show/384670/account-avatar-profile-user.svg',
    answers: {
      vthrdm985a262al8qx3do: 'optionOne',
      xj352vofupe1dqz9emx13r: 'optionTwo',
    },
    questions: ['loxhs1bqm25b708cmbf3g', 'vthrdm985a262al8qx3do'],
  },
  {
    id: 'mtsamis',
    password: 'xyz123',
    name: 'Mike Tsamis',
    avatarURL:
      'https://www.svgrepo.com/show/384674/account-avatar-profile-user-11.svg',
    answers: {
      xj352vofupe1dqz9emx13r: 'optionOne',
      vthrdm985a262al8qx3do: 'optionTwo',
      '6ni6ok3ym7mf1p33lnez': 'optionOne',
    },
    questions: ['6ni6ok3ym7mf1p33lnez', 'xj352vofupe1dqz9emx13r'],
  },
  {
    id: 'zoshikanlu',
    password: 'pass246',
    name: 'Zenobia Oshikanlu',
    avatarURL:
      'https://www.svgrepo.com/show/384676/account-avatar-profile-user-6.svg',
    answers: {
      xj352vofupe1dqz9emx13r: 'optionOne',
    },
    questions: [],
  },
];
const store = createStore((state = { users: mockUsers }, action) => state, {
  users: mockUsers,
});

describe('NoMatch', () => {
  it('matches the snapshot', () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NoMatch />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
