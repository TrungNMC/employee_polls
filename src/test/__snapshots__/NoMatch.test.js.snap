// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="no-macth"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
            >
              <img
                alt="logo"
                height="40"
                loading="lazy"
                src="logo.png"
                srcset="logo.png"
                sx="[object Object]"
                width="40"
              />
              <div
                class="MuiBox-root css-1lvtzne"
              >
                <button
                  aria-controls="menu-appbar"
                  aria-haspopup="true"
                  aria-label="account of current user"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="MenuIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root css-1t6c9ts"
              />
              <div
                class="MuiBox-root css-2uchni"
              >
                <button
                  aria-label="Open settings"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-17o22dy-MuiAvatar-root"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                          data-testid="PersonIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      class="MuiListItemText-root css-1trlqae-MuiListItemText-root"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </header>
        <img
          alt="Page Not Found"
          height="100%"
          loading="lazy"
          src="404.png"
          srcset="404.png"
          width="100%"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-10nezku-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="0"
        />
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-14ogsoa-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Login
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="no-macth"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <img
              alt="logo"
              height="40"
              loading="lazy"
              src="logo.png"
              srcset="logo.png"
              sx="[object Object]"
              width="40"
            />
            <div
              class="MuiBox-root css-1lvtzne"
            >
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-1t6c9ts"
            />
            <div
              class="MuiBox-root css-2uchni"
            >
              <button
                aria-label="Open settings"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-17o22dy-MuiAvatar-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                        data-testid="PersonIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiListItemText-root css-1trlqae-MuiListItemText-root"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
      <img
        alt="Page Not Found"
        height="100%"
        loading="lazy"
        src="404.png"
        srcset="404.png"
        width="100%"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
